src/WinGet.RestSource.IntegrationTest/Tests/Functions/PackageFunctionsTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="PackageFunctionsTests.cs" company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.WinGet.RestSource.IntegrationTest.Tests.Functions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.WinGet.RestSource.IntegrationTest.Common.Fixtures;
+    using Microsoft.WinGet.RestSource.IntegrationTest.Common.Helpers;
+    using Microsoft.WinGet.RestSource.IntegrationTest.Tests.Functions.TestData;
+    using Microsoft.WinGet.RestSource.Utils.Models.ExtendedSchemas;
+    using Microsoft.WinGet.RestSource.Utils.Models.Schemas;
+    using Xunit;
+
+    /// <summary>
+    /// Package functions integration tests.
+    /// </summary>
+    [Collection("IntegrationTestCollection")]
+    public class PackageFunctionsTests
+    {
+        private const string PowerToysPackageIdentifier = "Microsoft.PowerToys";
+        private IntegrationTestFixture fixture;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PackageFunctionsTests"/> class.
+        /// </summary>
+        /// <param name="fixture">An object of type <see cref="IntegrationTestFixture"/>.</param>
+        public PackageFunctionsTests(IntegrationTestFixture fixture)
+        {
+            this.fixture = fixture;
+        }
+
+        /// <summary>
+        /// Package function endpoint tests.
+        /// </summary>
+        /// <param name="helper">An object of type <see cref="PackageVersionTestHelper"/>.</param>
+        [Theory]
+        [ClassData(typeof(PackageTestData))]
+        public async void PackageFunctionEndpointTests(PackageVersionTestHelper helper)
+        {
+            Assert.NotNull(helper);
+            Assert.NotNull(helper.EndPointRequest);
+            Assert.NotNull(helper.PackageIdentifier);
+
+            await this.InitializeAsync(helper);
+
+            string url = $"{this.fixture.RestSourceUrl}/{helper.EndPointRequest.RelativeUrlPath.TrimStart('/')}";
+            var response = await IntegrationTestFixture.GetConsistentApiResponse<Package>(url);
+
+            if (helper.ApiResponse == PackageVersionTestHelper.ApiResponseScenario.IdentifierNotFound)
+            {
+                Assert.Null(response);
+            }
+
+            if (helper.ApiResponse == PackageVersionTestHelper.ApiResponseScenario.IdentifierFound)
```

Comment:
>PackageVersionTestHelper.ApiResponseScenario [](http://example.com/codeflow?start=38&length=44)

Mutually exclusive - maybe use switch? Then you could assert(false) for situations we know shouldn't happen.